src/api/controllers/socialAuth.js

```diff
@@ -0,0 +1,113 @@
+import model from '../models';
+import generateToken from '../../helpers/tokens/generate.token';
+
+const { User } = model;
+
+/**
+ *
+ *
+ * @class socialLogin
+ */
+class socialLogin {
+  /**
+   * signup || login via facebook
+   *
+   * @static
+   * @param {object} req - request object
+   * @param {object} res - response object
+   * @memberof socialLogin
+   * @returns {object} - the response body
+   */
+  static async facebookAuth(req, res) {
+    let existingUser = await User.findOne({
+      where: {
+        uniqueId: req.user.id
+      }
+    });
+    if (!existingUser) {
+      existingUser = await User.create({
+        username: req.user.name.familyName,
+        profileImage: req.user.photos[0].value,
+        provider: req.user.provider,
+        uniqueId: req.user.id
+      });
+    }
+    const tempUser = {
+      id: existingUser.dataValues.id,
+      username: existingUser.dataValues.username
+    };
+    const token = generateToken(tempUser, process.env.TOKEN_KEY);
+    return res.status(201).json({ user: { ...existingUser.get(), token } });
+  }
+
+  /**
+   * signup || login via Google
+   *
+   * @static
+   * @param {object} req - request object
+   * @param {object} res - request response
+   * @memberof socialLogin
+   * @returns {object} - the response body
+   */
+  static async googleAuth(req, res) {
+    let existingUser = await User.findOne({
+      where: {
+        uniqueId: req.user.id
+      }
+    });
+    if (!existingUser) {
+      existingUser = User.create({
+        username: req.user.displayName,
+        email: req.user.emails[0].value,
+        profileImage: req.user.photos[0].value,
+        provider: req.user.provider,
+        uniqueId: req.user.id
+      });
+      const tempUser = { username: req.user.displayName };
```

Comment:
@Ramadhan0 we need to pass an ID to the token generator.

Reply:
@Ramadhan0 Also, why do we need separate variables for `existingUser`, `tempUser` and `loginUser`? There seems to be a lot of repetition in these variables.